libutils/json.c

```diff
@@ -2749,6 +2749,62 @@ JsonParseError JsonParseFile(
     return JsonParseAnyFile(path, size_max, json_out, false);
 }
 
+bool JsonWalk(JsonElement *element,
+              JsonElementVisitor object_visitor,
+              JsonElementVisitor array_visitor,
+              JsonElementVisitor primitive_visitor,
+              void *data)
+{
+    assert(element != NULL);
+
+    if (element->type == JSON_ELEMENT_TYPE_PRIMITIVE)
+    {
+        if (primitive_visitor != NULL)
+        {
+            return primitive_visitor(element, data);
+        }
+        else
+        {
+            return true;
+        }
+    }
+    else
+    {
```

Comment:
Since the if branches always return, these `else` blocks are unnecessary. If you dropped them all the code below would be de-indented, and you would have a `return` at the end which is not inside an extra scope. I'd prefer that, but no big deal.

Reply:
Good point!